purchase_packaging/models/purchase_order_line.py

```diff
@@ -48,38 +48,48 @@ def _get_product_seller(self):
             uom_id=self.product_uom)
 
     @api.multi
-    @api.depends('product_purchase_uom_id', 'product_purchase_qty')
+    @api.depends('product_purchase_uom_id',
+                 'product_purchase_qty',
+                 'product_purchase_uom_id.category_id')
     def _compute_product_qty(self):
         """
         Compute the total quantity
         """
+        uom_categories = self.mapped("product_purchase_uom_id.category_id")
+        uom_obj = self.env['product.uom']
+        to_uoms = uom_obj.search(
+            [('category_id',
+              'in',
+              uom_categories.ids),
+             ('uom_type', '=', 'reference')])
+        uom_by_category = {to_uom.category_id: to_uom for to_uom in to_uoms}
         for line in self:
-            uom_obj = self.env['product.uom']
-            to_uom = uom_obj.search(
-                [('category_id',
-                  '=',
-                  line.product_purchase_uom_id.category_id.id),
-                 ('uom_type', '=', 'reference')], limit=1)
             if not line.product_purchase_uom_id:
```

Comment:
@Cedric-Pigeon @lmignon This is not correct as we must not stop computation. Maybe a self.filtered('product_purchase_uom_id') should be more convenient at the loop beginning.

Reply:
@rousseldenis good catch... should be 'continue"